viadot/tasks/sharepoint.py

```diff
@@ -0,0 +1,180 @@
+from typing import List
+import os
+import copy
+import pandas as pd
+from prefect import Task
+from prefect.utilities.tasks import defaults_from_attrs
+from prefect.utilities import logging
+from ..sources import Sharepoint
+
+logger = logging.get_logger()
+
+
+class SharepointToDF(Task):
+    """
+    Task for converting data from Sharepoint excel file to a pandas DataFrame.
+
+    Args:
+        path_to_file (str): Path to Excel file.
+        url_to_file (str):  Link to a file on Sharepoint.
+                        (e.g : https://{tenant_name}.sharepoint.com/sites/{folder}/Shared%20Documents/Dashboard/file). Defaults to None.
+        nrows (int, optional): Number of rows to read at a time. Defaults to 50000.
+        validate_excel_file (bool, optional): Check if columns in separate sheets are the same. Defaults to False.
+        if_empty (str, optional): What to do if query returns no data. Defaults to "warn".
+
+    Returns:
+        pd.DataFrame: Pandas data frame
+    """
+
+    def __init__(
+        self,
+        path_to_file: str = None,
+        url_to_file: str = None,
+        nrows: int = 50000,
+        validate_excel_file: bool = False,
+        if_empty: str = "warn",
+        *args,
+        **kwargs,
+    ):
+
+        self.if_empty = if_empty
+        self.path_to_file = path_to_file
+        self.url_to_file = url_to_file
+        self.nrows = nrows
+        self.validate_excel_file = validate_excel_file
+
+        super().__init__(
+            name="sharepoint_to_df",
+            *args,
+            **kwargs,
+        )
+
+    def __call__(self):
+        """Download Sharepoint data to a DF"""
+        super().__call__(self)
+
+    def check_column_names(
+        self, df_header: List[str] = None, header_to_compare: List[str] = None
+    ) -> List[str]:
+        """
+        Check if column names in sheets are the same.
+
+        Args:
+            df_header (List[str]): Header of df from excel sheet.
+            header_to_compare (List[str]): Header of df from previous excel sheet.
+
+        Returns:
+            list: list of columns
+        """
+        df_header_list = df_header.columns.tolist()
+        if header_to_compare is not None:
+            if df_header_list != header_to_compare:
+                logger.info("Columns in sheets are different")
+                exit()
```

Comment:
This is the wrong way to exit a program in python -- you should raise an exception here and specify the error message. The best practice is to use custom exceptions if applicable and I think it is here:

in `exceptions.py`:
```python
class ValidationError(Exception):
    pass
```
here:
```python
from ..exceptions import ValidationError

if df_header_list != header_to_compare:
    raise ValidationError("Columns in sheets are different")
```
Also, the message is not very specific, as you: a) don't specify that you meant column names only, b) don't tell the user which sheets are faulty, c) don't tell the user which columns caused the error. But we can let it be for now.